### DISCUSSION TITLE
How do i sync just one dir?

### DISCUSSION THREAD

**Discussion by spkgyk** [abraunegg/onedrive#1442] [Q&A] [answered]
I tried using onedrive --synchronize --single-directory Documents/Dir to sync just that directory, but today i look at my one drive folder on linux to find it has downloaded all 60GB of my onedrive storage. I only one to sync one directory, how do i do that? I dont want to delete the unwanted files locally because that will delete them on the server too. How do i remove all the unwanted local files and just keep one dir in sync? Many thanks

**Comment by abraunegg** [comment, marked as answer]
@spkgyk 
> I tried using onedrive --synchronize --single-directory Documents/Dir to sync just that directory, but today i look at my one drive folder on linux to find it has downloaded all 60GB of my onedrive storage.

Are you using the OneDrive client via the Ubuntu PPA? If yes, you are hit with 2 issues:
1. You are running an unsupported version - you must upgrade
2. You are impacted by https://github.com/abraunegg/onedrive/issues/1274 - you must fix this

Using `using onedrive --synchronize --single-directory Documents/Dir` is 100% the correct way to achieve what you are attempting. The issue you are facing is the above 2 points which will prevent your actual desired use of the client to work correctly.

Please check these 2 points first.
